File: Data Structures & Algorithms/validate-parentheses/submission-1.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stack = new Stack<>();
        for(int i=0; i<s.length(); i++){
            char targetChar = s.charAt(i);
            if(targetChar == '('
                ||targetChar == '{'
                ||targetChar == '['){
                    stack.push(targetChar);
            } else {
            // can swap to a switch case statement
                if(stack.size()==0){
                    return false;
                }
                char poppedChar = stack.pop();
                if(targetChar == ')') {
                    if(poppedChar != '(') {
                        return false;
                    }
                } else if(targetChar == '}') {
                    if(poppedChar != '{') {
                        return false;
                    }
                } else {
                    if(poppedChar != '[') {
                        return false;
                    }
                }
            }
        }
        if(stack.size()>0){
            return false;
        }
        return true;
    }
}

File: Data Structures & Algorithms/baseball-game/submission-1.py
class Solution:
    def calPoints(self, operations: List[str]) -> int:
        stack = []
        ret = 0
        for i in operations:
            if i == "+":
                a = stack[-2]
                b = stack[-1]
                c = a + b
                stack.append(c)
            elif i == "D":
                x = stack[-1] * 2
                stack.append(x)
            elif i == "C":
                stack.pop()
            else:
                stack.append(int(i))
        return sum(stack)